Basic documentation of object wrappers on its own page, with sample code

brodybits opened the pull request:
Keeps documentation of `Nan::ObjectWrapper` on its own page, along with examples to reflect the documentation at https://nodejs.org/api/addons.html:
- basic wrapped object
- wrapped object factory
- passing wrapped objects around

The examples are derived from the existing tests, with some adaptations and `using namespace` statement(s) removed. I tested the C++ code in the examples (with all `using namespace` statement(s) removed) to be sure that they will compile and run correctly. To be honest they are not so easy to understand for a beginner but I figure they would be better than having no samples.

kkoopa commented on doc/object_wrappers.md:
```diff
@@ -0,0 +1,247 @@
+
+  // Quick check:
+  if (maybe1.IsEmpty() || maybe2.IsEmpty()) {
+    info.GetReturnValue().Set(Nan::Undefined());
```
The default return value is undefined, so this whole line is unnecessary. On another note, it may also be slightly better to use the primitive setters directly e.g. `ReturnValue::SetUndefined`.

kkoopa commented on doc/object_wrappers.md:
```diff
@@ -0,0 +1,247 @@
+
+  static NAN_METHOD(New) {
+    if (info.IsConstructCall()) {
+      double value = info[0]->IsUndefined() ? 0 : info[0]->NumberValue();
```
Prefer `Nan::To<double>` over `NumberValue`

kkoopa commented on doc/object_wrappers.md:
```diff
@@ -0,0 +1,247 @@
+    SetPrototypeMethod(tpl, "getHandle", GetHandle);
+
+    constructor().Reset(tpl->GetFunction());
+    Nan::Set(target, Nan::New("MyObject").ToLocalChecked(), tpl->GetFunction());
```
`Nan::GetFunction(tpl).ToLocalChecked()`

kkoopa commented:
This is good to merge after those small nits have been fixed.

brodybits commented:
I made the changes and also made very similar changes to `test/cpp/objectwraphandle.cpp` to make it more future-proof as well.

kkoopa commented on doc/object_wrappers.md:
```diff
@@ -0,0 +1,248 @@
+    SetPrototypeMethod(tpl, "getHandle", GetHandle);
+
+    constructor().Reset(Nan::GetFunction(tpl).ToLocalChecked());
+    Set(target, Nan::New("MyObject").ToLocalChecked(),
```
This should have the namespace explicit, as the rest of the example does.

kkoopa commented on doc/object_wrappers.md:
```diff
@@ -0,0 +1,248 @@
+    v8::Local<v8::FunctionTemplate> tpl = Nan::New<v8::FunctionTemplate>(New);
+    tpl->InstanceTemplate()->SetInternalFieldCount(1);
+
+    SetPrototypeMethod(tpl, "getValue", GetValue);
```
namespace here too

brodybits commented:
My bad! I just fixed lines 70 and 131 to use the `Nan` namespace, and also added a comment at line 212 to mark the case where the return value is `undefined` by default.

kkoopa commented:
Very good. Thanks a lot.
